src/panels/OriginPanel/FocusView/FocusView.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  Button,
+  Container,
+  Divider,
+  Group,
+  Kbd,
+  Paper,
+  Space,
+  Text,
+  Title
+} from '@mantine/core';
+
+import { useGetStringPropertyValue, useOpenSpaceApi } from '@/api/hooks';
+import { FilterList } from '@/components/FilterList/FilterList';
+import { InfoBox } from '@/components/InfoBox/InfoBox';
+import { CancelIcon, FocusIcon } from '@/icons/icons';
+import { useAppSelector } from '@/redux/hooks';
+import { EngineMode, IconSize } from '@/types/enums';
+import { Identifier, PropertyOwner } from '@/types/types';
+import { NavigationAimKey, NavigationAnchorKey } from '@/util/keys';
+import { sgnUri } from '@/util/propertyTreeHelpers';
+
+import { RemainingFlightTimeIndicator } from '../RemainingFlightTimeIndicator';
+
+import { FocusEntry } from './FocusEntry';
+
+interface Props {
+  favorites: PropertyOwner[];
+  searchableNodes: PropertyOwner[];
+  heightFunction: (windowHeight: number) => number;
+  matcherFunction: (node: PropertyOwner, query: string) => boolean;
+}
+
+export function FocusView({
+  favorites,
+  searchableNodes,
+  heightFunction,
+  matcherFunction
+}: Props) {
+  const propertyOwners = useAppSelector((state) => state.propertyOwners.propertyOwners);
+  const engineMode = useAppSelector((state) => state.engineMode.mode);
+
+  const [anchor] = useGetStringPropertyValue(NavigationAnchorKey);
+  const [aim] = useGetStringPropertyValue(NavigationAimKey);
+
+  const luaApi = useOpenSpaceApi();
+
+  const anchorNode = anchor ? propertyOwners[sgnUri(anchor)] : undefined;
+  const isInFlight = engineMode === EngineMode.CameraPath;
+  const hasAim = aim !== '' && aim !== anchor;
+
+  function onSelect(
+    identifier: Identifier,
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ) {
+    const shouldRetarget = !event.shiftKey;
+    const shouldResetVelocities = !event.ctrlKey;
+    luaApi?.navigation.setFocus(identifier, shouldRetarget, shouldResetVelocities);
```

Comment:
This function is new and not in the `openspace-api-js.d.ts` file, so it leads to typescript and lint issues. 

Need help knowing what to do about it. @engbergandreas